.gitignore

```diff
@@ -5,3 +5,8 @@ Cargo.lock
 .vscode
 .idea
 *.o
+example/protocols/agent.rs
+example/protocols/empty.rs
+example/protocols/health.rs
+example/protocols/oci.rs
+example/protocols/types.rs
```

Comment:
xxx_ttrpc.rs should add to this file?

Reply:
I don't think ttrpc supports generate codes programmatically, so leave it alone.

Reply:
> @teawater @lifupan I have pushed the commit. In this commit I have renamed examples to example to make it a standalone cargo project, and I have created a file `example/build.rs` which will generate `example/protocols/*.rs` when users run `cargo run --example server/client` in the folder `example` the first time.

@Tim-Zhang
```
~/ttrpc-rust$ cargo run --example server
error: no example target named `server`

cargo build
    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
~/ttrpc-rust/example/target/debug$ ls
build  deps  examples  incremental
```
Could you tell me how to build this example?